File: frontend/src/types/review.ts
export type Severity = 'ERROR' | 'WARNING' | 'INFO'

export type ReviewType = 'all' | 'security' | 'performance' | 'bugs'

export interface PRMeta {
  pr_title?: string
  author_name: string
  author_avatar: string
  updated_at: string
  created_at?: string
  commits: number
  base_branch: string
  head_branch: string
  additions: number
  deletions: number
  files_changed: number
  files: { filename: string; additions: number; deletions: number }[]
}

export interface ReviewIssue {
  severity: Severity
  file: string
  line?: number
  position?: number
  title: string
  description: string
  suggestion?: string
  diff_snippet?: string
  confidence?: number
}

export interface WalkthroughEntry {
  file: string
  summary: string
}

export interface ReviewResult {
  pr_url: string
  summary: string
  risk_level: 'HIGH' | 'MEDIUM' | 'LOW'
  walkthrough: WalkthroughEntry[]
  issues: ReviewIssue[]
  stats: {
    files_changed: number
    additions: number
    deletions: number
    issues_by_severity: Record<Severity, number>
  }
  priority_files?: string[]
  risk_areas?: RiskArea[]
  merge_recommendation?: MergeRecommendation
}

export interface RiskArea {
  level: 'HIGH' | 'MEDIUM' | 'LOW'
  file: string
  title: string
  impact: string
}

export interface MergeRecommendation {
  decision: 'APPROVE' | 'REQUEST_CHANGES' | 'COMMENT'
  confidence: number
  reasons: string[]
}

export interface ReviewRequest {
  pr_url: string
  github_token?: string
  options?: {
    include_style: boolean
    context_lines: number
  }
}

File: backend/app/models/review.py
from enum import StrEnum

from pydantic import BaseModel


class Severity(StrEnum):
    ERROR = "ERROR"
    WARNING = "WARNING"
    INFO = "INFO"


class WalkthroughEntry(BaseModel):
    file: str
    summary: str


class ReviewIssue(BaseModel):
    severity: Severity
    file: str
    line: int | None = None
    position: int | None = None
    title: str
    description: str
    suggestion: str | None = None
    diff_snippet: str | None = None
    confidence: float = 1.0
    impact_type: str | None = None
    impact_statement: str | None = None


class ReviewStats(BaseModel):
    files_changed: int
    additions: int
    deletions: int
    issues_by_severity: dict[str, int] = {}


class RiskArea(BaseModel):
    level: str  # "HIGH" | "MEDIUM" | "LOW"
    file: str
    title: str
    impact: str


class MergeRecommendation(BaseModel):
    decision: str  # "APPROVE" | "REQUEST_CHANGES" | "COMMENT"
    confidence: int  # 0-100
    reasons: list[str]


class ReviewResult(BaseModel):
    pr_url: str
    summary: str
    risk_level: str
    walkthrough: list[WalkthroughEntry] = []
    issues: list[ReviewIssue]
    stats: ReviewStats
    priority_files: list[str] = []
    risk_areas: list[RiskArea] = []
    merge_recommendation: MergeRecommendation | None = None


class ReviewRequest(BaseModel):
    pr_url: str
    github_token: str | None = None
    perspective: str = "default"
    options: dict | None = None
    review_type: str = "all"
    model: str | None = None
    api_key: str | None = None
    base_url: str | None = None


class PostReviewRequest(BaseModel):
    pr_url: str
    github_token: str
    result: dict
